packages/framework/aqueduct/src/data-objects/pureDataObject.ts

```diff
@@ -73,10 +73,9 @@ export abstract class PureDataObject<I extends DataObjectTypes = DataObjectTypes
     public get handle(): IFluidHandle<this> { return this.innerHandle; }
 
     public static async getDataObject(runtime: IFluidDataStoreRuntime) {
-        const obj = (runtime as any)._dataObject as PureDataObject;
-        assert(obj !== undefined, 0x0bc /* "Runtime has no DataObject!" */);
-        await obj.finishInitialization(true);
-        return obj;
+        const obj = await runtime.entryPoint?.get();
```

Comment:
PureDataObject creates a handle in the constructor. Should it instead use the handle of the data store runtime now?

Reply:
Hmm interesting point. @anthony-murphy should we allow `initializeEntrypoint` to return either a `FluidObject` or an `IFluidHandle` and only wrap with our own handle instance in the first case? With the code as is, we end up with two separate handle instances that resolve to the same object: one is what the PureDataObject returns if we ask it directly for its `handle/IFluidHandle.handle` properties; the other one is what the data store runtime returns as its entrypoint. This sounds unintuitive and like it could result in objects not being GC'ed as expected if users "free" one of the handles but forget (or are unaware of) the other one.

I guess the same scenario (multiple handles) could happen as long as we potentially expect initializeEntrypoint to return a FluidObject; something else outside of the data store runtime could have created a handle for it, before or in the future. Is it something we should be concerned about, then?

I'll see about making PureDataObject return the entrypoint of the data store runtime instead of the handle it creates right now, to make sure that works; I have a feeling that it might run into a catch-22 issue.